exec/internal/compile/compile.go

```diff
@@ -276,7 +276,7 @@ func Compile(disassembly []disasm.Instr) ([]byte, *BytecodeMetadata) {
 			}
 
 			for _, table := range block.branchTables {
-				table.patchTable(table.blocksLen-depth-1, int64(block.offset))
+				table.patchTable(table.blocksLen-depth-1, int64(block.offset), inboundTargets)
```

Comment:
do we have an explicit test that exercizes this?

Reply:
Indirectly :) `TestSpec/testdata/spec/switch.wasm` has br_tables for dayz, and now that more instructions are supported (and hence more compilation candidates), the compiler was emitting code for instructions that the branch table was jumping into halfway. This change fixes that.